packages/fast-usdc-contract/src/exos/advancer.ts

```diff
@@ -190,86 +178,28 @@ export const prepareAdvancerKit = (
-            const { recipientAddress } = evidence.aux;
-            const decoded = decodeAddressHook(recipientAddress);
-            mustMatch(decoded, AddressHookShape);
-            if (decoded.baseAddress !== settlementAddress.value) {
-              throw Fail`⚠️ baseAddress of address hook ${q(decoded.baseAddress)} does not match the expected address ${q(settlementAddress.value)}`;
-            }
-            const { EUD } = decoded.query;
-            log(`decoded EUD: ${EUD}`);
-
-            // throws if it's neither CAIP-10 nor bare bech32.
-            const destination = chainHub.resolveAccountId(EUD);
-            const accountId = parseAccountId(destination);
-
-            // Dest must be a Cosmos account or support CCTP
-            if (
-              !(accountId.namespace === 'cosmos' || supportsCctp(destination))
-            ) {
-              const destChain = chainOfAccount(destination);
-              statusManager.skipAdvance(evidence, [
-                `Transfer to ${destChain} not supported.`,
-              ]);
-            }
-
-            const fullAmount = toAmount(evidence.tx.amount);
-            const { borrower, notifier, poolAccount } = this.state;
-            // do not advance if we've already received a mint/settlement
-            const mintedEarly = notifier.checkMintedEarly(
-              evidence,
-              destination,
-            );
-            if (mintedEarly) return;
-
-            // throws if requested does not exceed fees
-            const advanceAmount = feeTools.calculateAdvance(
-              fullAmount,
-              destination,
-            );
-
-            const { zcfSeat: tmpSeat } = zcf.makeEmptySeatKit();
-            // throws if the pool has insufficient funds
-            borrower.borrow(tmpSeat, advanceAmount);
-
-            // this cannot throw since `.isSeen()` is called in the same turn
-            statusManager.advance(evidence);
-
-            const depositV = localTransfer(
-              tmpSeat,
-              // @ts-expect-error LocalAccountMethods vs OrchestrationAccount
-              poolAccount,
-              harden({ USDC: advanceAmount }),
-            );
-            // WARNING: this must never reject, see handler @throws {never} below
-            // void not enforced by linter until #10627 no-floating-vows
-            void watch(depositV, this.facets.depositHandler, {
-              advanceAmount,
-              destination,
-              forwardingAddress: evidence.tx.forwardingAddress,
-              fullAmount,
-              tmpSeat,
-              txHash: evidence.txHash,
-            });
-          } catch (error) {
-            log('Advancer error:', error);
-            statusManager.skipAdvance(evidence, [(error as Error).message]);
-          }
+          const { notifier, borrower, poolAccount, settlementAddress } =
+            this.state;
+          // This synchronous function returns a Vow that does its own error handling.
```

Comment:
A function returning a vow is not quite "synchronous" in principle.

```suggestion
          // This function returns a Vow that does its own error handling.
```